Cabal-described/Cabal-described.cabal

```diff
@@ -19,7 +19,7 @@ library
     , QuickCheck
     , rere              >=0.1 && <0.3
     , tasty             <1.6
-    , tasty-quickcheck  <0.11
+    , tasty-quickcheck  <0.12
```

Comment:
The API changed, according to CI. Do we want to bump and conditionalize, or leave it at the old version?

Reply:
True... In particular, it looks like it's only one file?
```
tests/UnitTests/Distribution/Solver/Modular/QuickCheck/Utils.hs:35:24: error: [GHC-83865]
Error:     • Couldn't match expected type: (Test.QuickCheck.Random.QCGen, Int)
                  with actual type: Maybe a0
    • In the pattern: Nothing
      In the pattern: QuickCheckReplay Nothing
      In a case alternative:
          QuickCheckReplay Nothing -> getStdRandom random
   |
35 |       QuickCheckReplay Nothing -> getStdRandom random
   |                        ^^^^^^^
```
(and two more errors like that). In `tasty-quickcheck-0.11` [this constructor has changed](https://hackage.haskell.org/package/tasty-quickcheck-0.11/docs/Test-Tasty-QuickCheck.html#v:QuickCheckReplay). There's another constructor that appeared in 0.11, `QuickCheckReplayLegacy` that is close to the one we use (sans `Maybe`).

How could we conditionalize btw?  Leaving <0.11 would be fine, perhaps, but then every time we do `cabal outdated` (as per the wiki) it will make some noise -- I'm not a fan of that. I'd rather we had clean `cabal outdated`. In particular, if it's a matter of adding CPP in one tiny file in the test suite, I'd be fine with that.

Reply:
Yeh, by "conditionalize" I meant CPP on `MIN_VERSION_tasty_quickcheck` or whatever it is.

Reply:
@Kleidukos @Mikolaj opinions? Options with supporting `tasty-quickcheck` starting from 0.11 are:

1. add CPP
2. don't support it, stick to <0.11
3. support versions starting from 0.11. It looks like it's only a test dependency (but in the cabal-install package), so it's probably not too bad regarding the support window.

Reply:
It looks like https://github.com/UnkindPartition/tasty/pull/425 may provide the same functionality as that utils file, so we may be able to upgrade `tasty` and delete the whole file eventually.

Reply:
@ulysses4ever I pick option 3 :+1:

Reply:
@grayjay this is useful, thanks! But this doesn’t answer my question directly. Do you propose, as Hécate, to implement (3) (I.e. depend on newer versions of the package exclusively)?

Reply:
I think it's about the future, not now.

FWIW, I agree with @Kleidukos that it's reasonable to upgrade it because it's a test. That said, beware that we do run `validate.sh` on selected old GHC versions which may not be compatible with the new `tasty-quickcheck`.

Reply:
Yes obviously my pick is conditioned by "Does the CI pass" ;-)

Reply:
Yes, I think that we should start with (1) or (3), depending on what is required to satisfy the build, and then move towards (3), since `tasty-quickcheck` is only a test dependency.

Reply:
Hey all. I put ^>=0.11 for `tasty-quickcheck`, and CI seems to be fine with it. Check it out.

Reply:
Great! I verified that the QuickCheck seeds are being displayed correctly with the new API: https://github.com/haskell/cabal/actions/runs/10241795977/job/28330597202#step:19:17

Reply:
Oh, that's great, because I wasn't sure.